.circleci/config.yml

```diff
@@ -2,72 +2,77 @@ version: 2.1
 jobs:
   testGenerator:
     docker:
-      - image: circleci/node:10-browsers
+      - image: l.gcr.io/google/bazel
     working_directory: /home/circleci/project/gapic-generator-typescript
     steps:
       - checkout
-      - run:
-          name: Install protoc
-          command: |
-            wget https://github.com/protocolbuffers/protobuf/releases/download/v3.10.0/protoc-3.10.0-linux-x86_64.zip
-            cd / && sudo unzip /tmp/protoc-3.10.0-linux-x86_64.zip
-            protoc --version
-          working_directory: /tmp
-      - run:
-          name: Run npm install
-          command: |
-            npm install
       - run:
           name: Run generator unit tests
           command: |
-            npm link
-            npm test
-          environment:
-            NPM_CONFIG_PREFIX: /tmp/.npm-global
-      - run: 
-          name: Run linting
+            bazel test //:unit-tests
+      - run:
+          name: Unzip baseline tests output
           command: |
-            npm run lint
+            mkdir /home/circleci/project/baselines
+            unzip bazel-testlogs/unit-tests/test.outputs/outputs.zip -d /home/circleci/project/baselines
       - run:
-          name: copy protos to generated client library & move .test-out-* to project/out/
+          name: Copy protos to baselines
           command: |
-            for dir in .test-out-* ; do
+            for dir in /home/circleci/project/baselines/.test-out-* ; do
               cp -r test-fixtures/protos $dir
             done
-      - run: 
-          name: move test out folders to a separate folder
+      - run:
+          name: Copy compiled test applications
+          # bazel-bin/ is a symlink to /root/.cache/bazel/... and we won't have it
+          # available in Node container. Let's just save what we need.
           command: |
-            mkdir /home/circleci/project/baselines
-            cp -r .test-out-* /home/circleci/project/baselines
+            mkdir test-application-runners
+            cp bazel-bin/test/test-application/test-*.js test-application-runners/
+            cp bazel-testlogs/unit-tests/test.outputs/outputs.zip test-application-runners/
       - persist_to_workspace:
           root: /home/circleci/project
           paths:
             - gapic-generator-typescript
             - baselines
+  lint:
+    docker:
+      - image: circleci/node:10-browsers
```

Comment:
Maybe node.js 12?

Reply:
Well maybe yes!  (in a separate PR though - it's a separate Circle config change, this PR is already too big)